public/app/features/correlations/Forms/ConfigureCorrelationTargetForm.tsx

```diff
@@ -1,64 +1,171 @@
-              error={formState.errors.targetUID?.message}
+              label={t('correlations.target-form.type-label', 'Type')}
+              description={t('correlations.target-form.target-type-description', 'Specify the type of correlation')}
+              htmlFor="corrType"
+              invalid={!!errors.type}
             >
-              <DataSourcePicker
-                onChange={withDsUID(onChange)}
-                noDefault
-                current={value}
-                inputId="target"
-                width={32}
-                disabled={correlation !== undefined}
+              <Select
+                className={styles.typeSelect}
+                value={correlationType}
+                onChange={(value) => onChange(value.value)}
+                options={Object.values(CORR_TYPES_SELECT)}
+                aria-label="correlation type"
               />
             </Field>
           )}
         />
 
-        <QueryEditorField
-          name="config.target"
-          dsUid={targetUID}
-          invalid={!!formState.errors?.config?.target}
-          error={formState.errors?.config?.target?.message}
-        />
+        {correlationType === 'query' &&
+          (() => {
+            assertIsQueryTypeError(errors);
+            return (
+              <>
+                <Controller
+                  control={control}
+                  name="targetUID"
+                  rules={{
+                    required: {
+                      value: true,
+                      message: t('correlations.target-form.control-rules', 'This field is required.'),
+                    },
+                  }}
+                  render={({ field: { onChange, value } }) => (
+                    <Field
+                      label={t('correlations.target-form.target-label', 'Target')}
+                      description={t(
+                        'correlations.target-form.target-description-query',
+                        'Specify which data source is queried when the link is clicked'
+                      )}
+                      htmlFor="target"
+                      invalid={!!errors.targetUID}
+                      error={errors.targetUID?.message}
+                    >
+                      <DataSourcePicker
+                        onChange={withDsUID(onChange)}
+                        noDefault
+                        current={value}
+                        inputId="target"
+                        width={32}
+                        disabled={correlation !== undefined}
+                      />
+                    </Field>
+                  )}
+                />
+
+                <QueryEditorField
+                  name="config.target"
+                  dsUid={targetUID}
+                  invalid={!!errors?.config?.target}
+                  error={
+                    errors?.config?.target && 'message' in errors?.config?.target
+                      ? (errors?.config?.target as FieldError).message
```

Comment:
<img width="336" alt="Screenshot 2024-09-05 at 2 32 31 PM" src="https://github.com/user-attachments/assets/54b41e69-9691-42b2-8ceb-2825ef67fb29">

I'm not sure why, but the assert that maps all the fields to errors doesn't work with the `object` type I think. Unless there's something else I'm missing. I don't know how else to handle this without asserting the type here.

Reply:
is it possible that all the other properties are simple values? If it's trying to do a recursive mapping here then yeah property of type object should have been traversed into but then as it's not specified concretely it doesn't know what to do with it.

In any case probably not a big deal. Would just put a comment there for posterity that this is the case.